bcos-crypto/bcos-crypto/encrypt/HsmSM4Crypto.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ *  Copyright (C) 2021 FISCO BCOS.
+ *  SPDX-License-Identifier: Apache-2.0
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * @brief implementation for Hsm SM4 encryption/decryption
+ * @file HsmSM4Crypto.cpp
+ * @date 2022.11.04
+ * @author lucasli
+ */
+#include <bcos-crypto/encrypt/Exceptions.h>
+#include <bcos-crypto/encrypt/HsmSM4Crypto.h>
+#include <bcos-crypto/interfaces/crypto/CommonType.h>
+
+#include "CryptoProvider.h"
+#include "sdf/SDFCryptoProvider.h"
+using namespace hsm;
+using namespace hsm::sdf;
+//using namespace bcos;
+using namespace bcos::crypto;
+
+bcos::bytesPointer bcos::crypto::HsmSM4Encrypt(const unsigned char* _plainData, size_t _plainDataSize,
+    const unsigned char* _key, size_t, const unsigned char* _ivData, size_t)
```

Comment:
why there are tow unused size_t parameters?

Reply:
because base class SymmetricEncryption have this parameters but that wasn't not uesd in hsm, just for API compatibility